rust/worker/src/blockstore/arrow/provider.rs

```diff
@@ -165,19 +196,19 @@ impl BlockManager {
         block.to_block_delta::<KeyWrite::ReadableKey<'new>, ValueWrite::ReadableValue<'new>>(delta)
     }
 
-    pub(super) fn commit<K: ArrowWriteableKey, V: ArrowWriteableValue>(&self, delta: &BlockDelta) {
+    pub(super) fn commit<K: ArrowWriteableKey, V: ArrowWriteableValue>(
+        &self,
+        delta: &BlockDelta,
+    ) -> Block {
         let record_batch = delta.finish::<K, V>();
         let block = Block::from_record_batch(delta.id, record_batch);
-        self.read_cache.write().insert(block.id, block);
+        block
     }
 
     pub(super) async fn get(&self, id: &Uuid) -> Option<Block> {
-        let block = {
-            let cache = self.read_cache.read();
-            cache.get(id).cloned()
-        };
+        let block = self.block_cache.get(id);
         match block {
-            Some(block) => Some(block),
+            Some(block) => Some(block.clone()),
```

Comment:
The diff here looks wrong - we lost a bunch of tracing we have on main. 

https://github.com/chroma-core/chroma/blob/2ae46d2dcdea1e57914dc8a3c68181840452eecb/rust/worker/src/blockstore/arrow/provider.rs#L191

I think something went wrong?